src/base/static/js/views/place-detail-view.js

```diff
@@ -45,13 +78,61 @@
         datasetId: self.options.datasetId
       });
 
+      // fetch comments here instead of in render(), to avoid fetching on
+      // a re-render and possibly conflicting with in-progress update/delete calls
+      this.model.submissionSets[this.surveyType].fetchAllPages();
+
+      this.model.attachmentCollection.on("add", this.onAddAttachmentWrapper, this);
+
       this.$el.on('click', '.share-link a', function(evt){
 
         // HACK! Each action should have its own view and bind its own events.
         var shareTo = this.getAttribute('data-shareto');
 
         Util.log('USER', 'place', shareTo, self.model.getLoggingDetails());
       });
+
+      this.buildFieldListForRender();
+    },
+
+    setUrlTitlePrefix: function() {
+      var self = this, 
+      layer = _.find(this.options.mapConfig.layers, function(layer) {
+        return layer.id === self.model.get("datasetId");
+      });
+
+      this.$el.find(".url-prefix")
+        .html(
+          window.location.protocol + "//" + window.location.hostname + "/"
+        );
+
+      this.$el.find(".url-readout")
+        .html(Util.getUrl(this.model));
+    },
+
+    onUpdateUrlTitle: function(evt) {
+      var urlReadout = $(evt.currentTarget)
+        .siblings(".url-readout-container")
+        .find(".url-readout");
+
+      if ($(evt.currentTarget).val() === "") {
+        urlReadout.html(Util.getShareaboutsUrl(this.model));
+      } else {
+        urlReadout.html(Util.prepareCustomUrl($(evt.currentTarget).val()));
+      }
+    },
+
+    onAddAttachmentWrapper: function(attachment) {
+      this.onAddAttachmentCallback.call(this.onAddAttachmentCallbackContext, attachment);
+    },
+
+    saveDraftChanges: function() {
+      var attrs = this.scrapeForm();
+      Util.localstorage.save(this.LOCALSTORAGE_KEY, attrs, 30) // save for 30 days
```

Comment:
30 days is arbitrary. I was worried that indefinite storage could clog up `localstorage`, but maybe this is not a big concern?